### PR TITLE
Build Web version

### PR DESCRIPTION
**Pull request by sherlockvn**
https://github.com/linagora/twake-on-matrix/assets/43041967/cae7775a-cb2b-4465-a176-10542d2c7cfc

### CODE REVIEW

File: lib/pages/chat/chat_view.dart [hunk 1]
```diff
@@ -128,15 +128,32 @@ class ChatView extends StatelessWidget {
...
+            margin: EdgeInsets.symmetric(
+              vertical: ChatViewStyle.paddingVerticalActionButtons,
+              horizontal: 6.0,
+            ),
           ),
+          if (FluffyThemes.isColumnMode(context))
+            TwakeIconButton(
+              onPressed: () {print;},
+              icon: Icons.search,
+              tooltip: "Search",
```
Comment (nqhhdev) [inline lib/pages/chat/chat_view.dart:150]:
```suggestion
              tooltip: L10n.of(context)!.search,
```

File: lib/pages/chat/chat_view.dart [hunk 2]
```diff
@@ -197,8 +214,10 @@ class ChatView extends StatelessWidget {
                 appBar: AppBar(
                   toolbarHeight: 64,
                   surfaceTintColor: Colors.transparent,
-                  leadingWidth: 8 + 24 + 8,
-                  leading: controller.selectMode
+                  leadingWidth: FluffyThemes.isColumnMode(context) ? 16 : 40,
```
Comment (nqhhdev) [inline lib/pages/chat/chat_view.dart:217]: IMO: move to ChatViewStyle

File: lib/pages/chat_list/home_screen.dart [hunk 3]
```diff
@@ -57,6 +47,22 @@ class _HomeScreenState extends State<HomeScreen> {
...
+  void onDestinationSelected({
+    required int selected, 
+    required BuildContext context
+  }) {
+    setState(() {
+      selectedIndex = selected;
+      VRouter.of(context).to(BottomTabbar.fromIndex(selected).path);
+      if (selected == BottomTabbar.stories.tabIndex) {
+        Fluttertoast.showToast(
+          msg: "Stories is not ready yet!",
```
Comment (nqhhdev) [inline lib/pages/chat_list/home_screen.dart:59]: IMO: Move to `L10n `

File: lib/pages/new_private_chat/widget/search_contact_appbar.dart [hunk 4]
```diff
@@ -56,13 +57,16 @@ class _SearchContactAppBarState extends State<SearchContactAppBar> {
...
-      paddingAll: 8.0,
-      margin: const EdgeInsets.symmetric(horizontal: 8.0, vertical: 12.0),
-    ),
+    leadingWidth: FluffyThemes.isColumnMode(context) ? 16 : 56,
+    leading: FluffyThemes.isColumnMode(context)
+      ? const SizedBox(width: 16.0,)
+      : TwakeIconButton(
+        icon: Icons.arrow_back,
+        onPressed: () => VRouter.of(context).pop(),
+        tooltip: "Back",
```
Comment (nqhhdev) [inline lib/pages/new_private_chat/widget/search_contact_appbar.dart:66]:
```suggestion
        tooltip: "L10n.of(context)!.back",
```

File: lib/widgets/layouts/scaffold_view_web.dart [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+    return Scaffold(
+      appBar: AppBar(
+        toolbarHeight: 76,
+        leadingWidth: 300,
+        leading: Row(
+          children: [
+            const SizedBox(width: 16.0,),
+            SvgPicture.asset(ImagePaths.icTwakeImageLogo, width: 44),
+            const SizedBox(width: 16.0,),
+            SvgPicture.asset(ImagePaths.icTwakeLogo, height: 30, fit: BoxFit.fitHeight,),
```
Comment (nqhhdev) [inline lib/widgets/layouts/scaffold_view_web.dart:39]: `,` not cleannnnn =))))

File: lib/widgets/layouts/scaffold_view_web.dart [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+          ],
+        ),
+        actions: [
+          TwakeIconButton(
+            margin: const EdgeInsets.symmetric(horizontal: 8.0),
+            tooltip: 'Apps',
```
Comment (nqhhdev) [inline lib/widgets/layouts/scaffold_view_web.dart:45]: Move to `L10n`

File: lib/widgets/layouts/scaffold_view_web.dart [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+  void onDestinationSelected({
+    required int selected, 
+    required BuildContext context
+  }) {
+    setState(() {
+      selectedIndex = selected;
+      VRouter.of(context).to(BottomTabbarWeb.fromIndex(selected).path);
+      if (selected == BottomTabbarWeb.stories.tabIndex) {
+        Fluttertoast.showToast(
+          msg: "Stories is not ready yet!",
```
Comment (nqhhdev) [inline lib/widgets/layouts/scaffold_view_web.dart:119]: Move to `L10n`

**Review by nqhhdev — APPROVED** [review]

**Review by imGok — APPROVED** [review]

File: lib/widgets/layouts/two_column_layout_web.dart [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+  const TwoColumnLayoutWeb({
+    super.key,
+    required this.mainView,
+    required this.sideView,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      padding: const EdgeInsets.only(left: 0, right: 16.0, top: 16.0, bottom: 16.0),
```
Comment (dab246) [inline lib/widgets/layouts/two_column_layout_web.dart:19]: **Recommended**: Should use `EdgeInsetsDirectional` to support later `RTL` languages

File: lib/data/hive/hive_collection_tom_database.dart [hunk 9]
```diff
@@ -29,7 +30,11 @@ class HiveCollectionToMDatabase {
     HiveAesCipher? hiverCipher;
     try {
       // Workaround for secure storage is calling Platform.operatingSystem on web
-      if (kIsWeb) throw MissingPluginException();
+      if (kIsWeb) {
+        // ignore: unawaited_futures
+        html.window.navigator.storage?.persist();
```
Comment (hoangdat) [inline lib/data/hive/hive_collection_tom_database.dart:35]: @dab246 also questioned about it. Can you explain something?

Comment (dab246) [reply lib/data/hive/hive_collection_tom_database.dart:35]: If you call this you want your app not to be automatically cleared when storage is low and you can just remove it manually by going to browser settings. 
By default, a site's data `(IndexedDB, Cache API,...)` falls into the best effort category, the browser may evict site data at its discretion, for example when device storage is low.

**Pull request merged**